dicodile/update_z/distributed_sparse_encoder.py

```diff
@@ -30,13 +32,25 @@ def __init__(self, n_workers, w_world='auto', hostfile=None, verbose=0):
         self.hostfile = hostfile
         self.verbose = verbose
 
-    def init_workers(self, X, D_hat, reg, params, z0=None, DtD=None):
+    def init_workers(self, X, D_hat, reg, params, z0=None,
+                     DtD=None, rank1=False, n_channels=0):  # XXX
+
+        self.rank1 = rank1
+
+        if rank1:
+            assert n_channels > 0, \
+                "n_channels is required to compute rank-1 encoding"
+            uv_hat = D_hat
+            D_hat = get_D(D_hat, n_channels)
 
         # compute the partition for the signals
         assert D_hat.ndim - 1 == X.ndim, (D_hat.shape, X.shape)
         n_channels, *sig_support = X.shape
         n_atoms, n_channels, *atom_support = self.D_shape = D_hat.shape
 
+        if rank1:
+            self.uv_shape = D_hat.shape
```

Comment:
are we using this anywhere?

Reply:
Just for the assertion (ensuring this does not change from one call of `set_worker_D` to the other)

Reply:
ok